game-core/src/main/java/games/strategy/triplea/ui/menubar/HelpMenu.java

```diff
@@ -192,49 +190,15 @@ private static String getUnitImageUrl(final UnitType unitType, final PlayerId pl
   private void addUnitHelpMenu() {
     final String unitHelpTitle = "Unit Help";
     final JMenuItem unitMenuItem = add(SwingAction.of(unitHelpTitle, e -> {
-      final JEditorPane editorPane = new JEditorPane();
+      final Result<String> result = Interruptibles.awaitResult(() -> GameRunner.newBackgroundTaskRunner()
+          .runInBackgroundAndReturn("Calculating Data", () -> getUnitStatsTable(gameData, uiContext)));
+      final JEditorPane editorPane = new JEditorPane("text/html",
+          result.result.orElse("Failed to calculate Data"));
       editorPane.setEditable(false);
-      editorPane.setContentType("text/html");
-      editorPane.setText(getUnitStatsTable(gameData, uiContext));
       editorPane.setCaretPosition(0);
       final JScrollPane scroll = new JScrollPane(editorPane);
       scroll.setBorder(BorderFactory.createEmptyBorder());
-      final Dimension screenResolution = Toolkit.getDefaultToolkit().getScreenSize();
-      // not only do we have a start bar, but we also have the message dialog to account for just the scroll bars plus
-      // the window sides
-      final int availHeight = screenResolution.height - 120;
-      final int availWidth = screenResolution.width - 40;
-      scroll
-          .setPreferredSize(new Dimension(
-              (scroll.getPreferredSize().width > availWidth ? availWidth
-                  : (scroll.getPreferredSize().height > availHeight
-                      ? Math.min(availWidth, scroll.getPreferredSize().width + 22)
-                      : scroll.getPreferredSize().width)),
-              (scroll.getPreferredSize().height > availHeight ? availHeight
-                  : (scroll.getPreferredSize().width > availWidth
-                      ? Math.min(availHeight, scroll.getPreferredSize().height + 22)
-                      : scroll.getPreferredSize().height))));
```

Comment:
Looks like you removed all this custom height/width logic. Is it just not needed?

Reply:
I have tried with and without an couldn't spot a difference.
I can do a side to side comparison if you're not convinced

Reply:
Ok, I'm good merging this and doing some testing with the pre-release.